lib/rubocop/cop/style/option_hash.rb

```diff
@@ -0,0 +1,55 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for options hashes and discourages them if the
+      # current Ruby version supports keyword arguments
+      #
+      # @example
+      #   Instead of:
+      #
+      #   def fry(options = {})
+      #     temperature = options.fetch(:temperature, 300)
+      #     ...
+      #   end
+      #
+      #   Prefer:
+      #
+      #   def fry(temperature: 300)
+      #     ...
+      #   end
+      class OptionHash < Cop
+        MSG = 'Prefer keyword arguments to options hashes.'
+
+        def on_args(node)
+          return unless supports_keyword_arguments?
+
+          last_arg = node.to_a.last
+
+          # asserting that there was an argument at all
+          return unless last_arg
+
+          # asserting last argument is an optional argument
+          return unless last_arg.optarg_type?
+
+          default_value = last_arg.to_a.last
+
+          # asserting default value is a hash
+          return unless default_value.hash_type?
+
+          # asserting default value is empty hash
+          return unless default_value.to_a.length == 0
```

Comment:
I'm not sure how to write this as a splat because I don't know how many key value pairs the hash will have